staging/src/k8s.io/client-go/tools/cache/shared_informer.go

```diff
@@ -180,6 +197,29 @@ type SharedInformer interface {
 	// The handler should return quickly - any expensive processing should be
 	// offloaded.
 	SetWatchErrorHandler(handler WatchErrorHandler) error
+
+	// IsStopped reports whether the informer has already been stopped.
+	// Adding event handlers to already stopped informers is not possible.
+	// An informer already stopped will never be started again.
+	IsStopped() bool
+}
+
+// ResourceEventHandlerHandle is a handle returned by the
+// registration methods of SharedInformers for a registered
+// ResourceEventHandler. It can be used later on to remove
+// a registration again.
+// This indirection is required, because the ResourceEventHandler
+// interface can be implemented by non-go-comparable handlers, which
+// could not be removed from a list anymore.
+type ResourceEventHandlerHandle struct {
+	listener *processorListener
```

Comment:
I expected to find a key to a map here :) (I haven't read the implementation)

Reply:
I made it constant now and switched to a value return for the handle. 

The problem basically was something different. If a handle was removed from an informer it was NOT registered for,
it got corrupted. Now you can remove a handle from any informer without any harm.
The operation is only executed, if the handle is still active for the informer it is passed to remove it.

Reply:
IMO, no, the removal should always succeed (i.e., be safe to call twice). This makes it very convenient if a user has multiple shutdown paths.